packages/internal/src/__tests__/validateSchemaForDirectiveAuthErrors.test.ts

```diff
@@ -0,0 +1,85 @@
+import path from 'path'
+
+import { CodeFileLoader } from '@graphql-tools/code-file-loader'
+import { loadTypedefs } from '@graphql-tools/load'
+import { mergeTypeDefs } from '@graphql-tools/merge'
+import { DocumentNode } from 'graphql'
+
+import { getPaths } from '../,,/../paths'
+import { validateSchemaForDirectives } from '../validateSchema'
+
+const FIXTURE_ERROR_PATH = path.resolve(
+  __dirname,
+  '../../../../__fixtures__/example-todo-main-with-errors'
+)
+
+const projectTypeSrc = async (sdlFile: string) =>
+  await loadTypedefs([`graphql/**/${sdlFile}.sdl.{js,ts}`], {
+    loaders: [
+      new CodeFileLoader({
+        noRequire: true,
+        pluckConfig: {
+          globalGqlIdentifierName: 'gql',
+        },
+      }),
+    ],
+    cwd: getPaths().api.src,
+  })
+
+const validateSdlFile = async (sdlFile: string): Promise<boolean> => {
+  let isSdlValid
+
+  const projectTypeSrcFiles = await projectTypeSrc(sdlFile)
+
+  try {
+    // The output of the above function doesn't give us the documents directly
+    const projectDocumentNodes = Object.values(projectTypeSrcFiles)
+      .map(({ document }) => document)
+      .filter((documentNode): documentNode is DocumentNode => {
+        return !!documentNode
+      })
+
+    // Merge in the rootSchema with JSON scalars, etc.
+    const mergedDocumentNode = mergeTypeDefs([projectDocumentNodes])
+
+    validateSchemaForDirectives(mergedDocumentNode)
+    isSdlValid = true
```

Comment:
I think these tests need to be more specific, since we have more than one error in validate now!

Reply:
Tests now check for specific error message when validating SDLs for auth -- both individual and all together.